### ISSUE TITLE
Wait for IP to be reported

### ISSUE THREAD

**Issue by zdzielinski** [oVirt/terraform-provider-ovirt#281]
Currently, if one desires to define a provisioner for a VM that sets a network interface with an expected IP address, they must obtain this information from a data source and wait for it to propagate (one solution is to use `time_sleep`, not the best of practices) before defining something along the lines of a `null_resource` to provision a VM once it's network information has propagated.

It would be extremely nice to have this provider handle that data propagation for us, and also handle the cases where that data has not propagated successfully, and showing us an error. Adding this in would make using this module much more seamless, as we could theoretically add provisions straight onto VM resource declarations instead of waiting on propagation, obtaining data, and creating a separate provisioning section.

I'm not sure how possible this would be - but comparing this provider to that of, for example; the AWS provider - one is able to define a VM and add a provisioner directly into an AWS resource declaration as it exports attributes that contain its IP address and network information (and, I assume, it also waits for those attributes to exist before continuing).

**Comment by emesika** [comment]
@zdzielinski 

can you please specify exactly (in pseudo code) what is the current behavior and  what is the behavior you are expecting to make that more clear ?

**Comment by janosdebugs** [comment]
Hey @zdzielinski,

as we are rewriting the Terraform provider, I did a little investigation. If you have a recent oVirt version and have the guest agent installed, the IP address will be reported back to oVirt and we can extract it from the API.

To that end, I will be adding a resource or data source which lets you wait for that IP to be reported and lets you add a provisioner. Pseudocode:

```terraform
resource "ovirt_wait_for_vm_ip" "myip" {
    vm_id = ovirt_vm.myvm.id
    device = "eth0"

    provisioner "remote-exec" {
        inline = [
            "...",
        ]
    }
}
```

I hope this helps.
